src/mbgl/util/chrono.cpp

```diff
@@ -22,8 +22,8 @@ std::string rfc1123(Timestamp timestamp) {
     std::time_t time = std::chrono::system_clock::to_time_t(timestamp);
     std::tm info;
     _gmtime(&time, &info);
-    char buffer[30];
-    snprintf(buffer, 30, "%s, %02d %s %4d %02d:%02d:%02d GMT", week[info.tm_wday], info.tm_mday,
+    char buffer[32];
```

Comment:
could you add a comment, explaining why `30` does not suffice?

Reply:
The complete warning/error message is like this:

../src/mbgl/util/chrono.cpp:26:26: error: ‘__builtin___snprintf_chk’ output may be truncated before the last format character [-Werror=format-truncation=]
snprintf(buffer, 30, "%s, %02d %s %4d %02d:%02d:%02d GMT", week[info.tm_wday], info.tm_mday,
^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from /usr/include/stdio.h:867,
from /usr/include/c++/8/cstdio:42,
from /usr/include/c++/8/ext/string_conversions.h:43,
from /usr/include/c++/8/bits/basic_string.h:6400,
from /usr/include/c++/8/string:52,
from ../include/mbgl/util/chrono.hpp:4,
from ../src/mbgl/util/chrono.cpp:1:
/usr/include/x86_64-linux-gnu/bits/stdio2.h:67:35: note: ‘__builtin___snprintf_chk’ output 24 or more bytes (assuming 31) into a destination of size 30
return __builtin___snprintf_chk (__s, __n, __USE_FORTIFY_LEVEL - 1,
~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
__bos (__s), __fmt, __va_arg_pack ());
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

Actually, assuming the year has 4 digits, if one counts the charactors, it's 29 of them. Plus the teminating '\0', 30 should be enough.
However, the compiler thinks the year might beyond 4 digits. I've given 2 extra bytes in the fix. In theory, it's still not enough. However, the compiler doesn't complain any more and we'll probably not see the year which can overflow. :-)
Another observation is that the compiler option "-Os" plays a role here as well. I've tried a standalone test cpp and use the compiler option coming from "cmake -GNinja -DCMAKE_BUILD_TYPE=RelWithDebInfo". I can repro the issue. However, when I remove the "-Os" option, the warning disappears.

Anyway, having the extra 2 bytes on the stack is harmless and solves my problem.

Reply:
Do you mean to add a comment in the code or just comment on github?

Reply:
just short comment in the code describing the problem and the compiler version.

Reply:
I've updated my commit. Please check.